src/main/utility/mod.rs

```diff
@@ -268,37 +277,35 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_tilde_expansion() {
+    fn test_resolve_exec_path() {
+        fn resolve_exec(path: &str) -> anyhow::Result<std::path::PathBuf> {
+            resolve_path(Path::new(path))
+        }
         if let Ok(ref home) = std::env::var("HOME") {
+            assert_eq!(resolve_exec("bash").unwrap(), which::which("bash").unwrap());
+
             assert_eq!(
-                tilde_expansion("~/test"),
+                resolve_exec("~/test").unwrap(),
                 [home, "test"].iter().collect::<std::path::PathBuf>()
             );
 
             assert_eq!(
-                tilde_expansion("~"),
+                resolve_exec("~").unwrap(),
                 [home].iter().collect::<std::path::PathBuf>()
             );
 
             assert_eq!(
-                tilde_expansion("~/"),
+                resolve_exec("~/").unwrap(),
                 [home].iter().collect::<std::path::PathBuf>()
             );
 
             assert_eq!(
-                tilde_expansion("~someuser/test"),
-                ["/home", "someuser", "test"]
-                    .iter()
-                    .collect::<std::path::PathBuf>()
-            );
```

Comment:
If we don't want to support this syntax, I think we should raise an error rather than return an incorrect path.